src/axom/multimat/multimat.cpp

```diff
@@ -382,16 +460,53 @@ void MultiMat::setCellMatRel(const vector<bool>& vecarr, DataLayout layout)
     counts[i] = cardinality;
   }
 
+  if(AllocatorOnDevice(m_slamAllocatorId))
+  {
+    // Copy counts and indices to GPU memory.
+    counts = IndBufferType(counts, m_slamAllocatorId);
+    indices = IndBufferType(indices, m_slamAllocatorId);
+  }
+
   setCellMatRel(counts, indices, layout);
 }
 
+template <typename ExecSpace, typename IndexType>
+void ScanRelationOffsetsRAJA(const axom::ArrayView<const IndexType> counts,
+                             const axom::ArrayView<IndexType> begins,
+                             const axom::ArrayView<IndexType> firstIndices)
```

Comment:
Note: The `firstIndices` view is marked as `const` but the underlying array is modified by this function

Reply:
`ArrayView<T>` is a shallow-const type, so `ArrayView<const IndexType>` is read-only while `ArrayView<IndexType>` is modifiable. The `const` in front of both is just to work around the GCC 8.1.0 issue with lambda-capture of `ArrayView`s.